enhancements/microshift/microshift-router-configuration-performance.md

```diff
@@ -0,0 +1,196 @@
+| tuningOptions -> headerBufferMaxRewriteBytes |  | Yes | ROUTER_MAX_REWRITE_SIZE | 8192 |  |
+| tuningOptions -> healthCheckInterval |  | Yes | ROUTER_BACKEND_CHECK_INTERVAL | 5000ms |  |
+| tuningOptions -> clientTimeout |  | Yes | ROUTER_DEFAULT_CLIENT_TIMEOUT | 30s |  |
+| tuningOptions -> clientFinTimeout |  | Yes | ROUTER_CLIENT_FIN_TIMEOUT | 1s |  |
+| tuningOptions -> serverTimeout |  | Yes | ROUTER_DEFAULT_SERVER_TIMEOUT | 30s |  |
+| tuningOptions -> serverFinTimeout |  | Yes | ROUTER_DEFAULT_SERVER_FIN_TIMEOUT | 1s |  |
+| tuningOptions -> tunnelTimeout |  | Yes | ROUTER_DEFAULT_TUNNEL_TIMEOUT | 1h |  |
+| tuningOptions -> tlsInspectDelay |  | Yes | ROUTER_INSPECT_DELAY | 5s |  |
+| tuningOptions -> threadCount |  | No | ROUTER_THREADS | 4 |  |
+| tuningOptions -> maxConnections |  | Yes | ROUTER_MAX_CONNECTIONS | 50000 |  |
+| LogEmptyRequests |  | Yes | ROUTER_DONT_LOG_NULL |  |  |
+| HTTPEmptyRequestsPolicy |  | Yes | ROUTER_HTTP_IGNORE_PROBES |  |  |
+| HTTP2 |  | NO | ROUTER_DISABLE_HTTP2 | true | Bool |
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+    - ...
+```
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
+    - ...
+```
+
+For more information check each individual section.
+
+### Topology Considerations
+TBD
+
+#### Standalone Clusters
+N/A
+
+#### Single-node Deployments or MicroShift
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+
+
+### Risks and Mitigations
+TBD
+
+### Drawbacks
+
+- Setting the timeout higher may cause some dead connections to be kept open
+  for longer, and would add to the memory footprint of the router
+- Setting the timeout too low can cause connection closure before the server or
+  client has enough time to respond
+
+
+## Design Details
+TBD
+
+## Open Questions
+N/A
+
+## Test Plan
+All of the changes listed here will be included in the current e2e scenario
+testing harness in MicroShift.
+
+## Graduation Criteria
+Targeting GA for MicroShift 4.18 release.
+
+### Dev Preview -> Tech Preview
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+
+### Tech Preview -> GA
+- More testing (upgrade, downgrade)
+- Sufficient time for feedback
+- Available by default
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+### Removing a deprecated feature
+N/A
+
+## Upgrade / Downgrade Strategy
```

Comment:
Are these only for minor-version updates/downgrades? What about z-stream updates?